app/imports/components/fields/signatureField.jsx

```diff
@@ -0,0 +1,76 @@
+/* globals SignaturePad */
+import React, { Component } from 'react';
+
+import { connectField } from 'uniforms';
+
+class SignatureComponent extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount() {
+    this.canvas = this.refs.signatureCanvas;
+    this.canvasContext = this.canvas.getContext('2d');
+
+    this.signaturePad = new SignaturePad(this.canvas);
+
+    window.addEventListener('resize', this.handleResize.bind(this));
+
+    this.drawSignature();
+
+  }
+
+  componentDidUpdate() {
+    this.drawSignature();
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize.bind(this));
+  }
+
+  drawSignature() {
+    // If signature is empty just clear the canvas
+    if (this.props.value === '') {
+      return this.signaturePad.clear();
+    }
+
+    let img = new Image();
+
+    img.onload = function() {
+      this.canvasContext.drawImage(img,0,0,img.width,img.height,0,0,this.canvas.width,this.canvas.height);
+    }.bind(this);
+
+    img.src = this.props.value;
+  }
+
+  clearSignature() {
+    // maybe use a confirm() or something here
+    this.props.onChange('');
+  }
+
+  handleMouseUp() {
```

Comment:
there is a problem with using this event. if you draw a sig and fall outside the box before mouseup it still draws the lines but it doesnt save the data. Not sure the clear solution to this but def a bug, I guess you need an event hook to call the data save, but it needs to be more than just this event. Is there no canvas on change event?